sdk/core/azure-core/test/ut/curl_session_test.cpp

```diff
@@ -28,7 +28,7 @@ namespace Azure { namespace Core { namespace Test {
     EXPECT_CALL(*curlMock, SendBuffer(_, _, _)).WillOnce(Return(CURLE_OK));
     EXPECT_CALL(*curlMock, ReadFromSocket(_, _, _))
         .WillOnce(DoAll(
-            SetArrayArgument<1>(response.data(), response.data() + response.size()),
+            SetArrayArgument<0>(response.data(), response.data() + response.size()),
```

Comment:
@vhvb1989 please verify this mocking framework update makes sense.

Reply:
looks good, but I still don't like this change to Core.
If @RickWinter  and @JeffreyRichter  want it, that's fine anyway.
Consider that keeping the Context as the last argument for all these APIs will require adding any new parameter in the middle